voice/src/main/kotlin/Streams.kt

```diff
@@ -0,0 +1,69 @@
+package dev.kord.voice
+
+import dev.kord.common.annotation.KordVoice
+import dev.kord.common.entity.Snowflake
+import dev.kord.voice.gateway.Speaking
+import dev.kord.voice.gateway.on
+import dev.kord.voice.udp.AudioPacket
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.*
+import kotlin.coroutines.CoroutineContext
+
+@KordVoice
+class Streams(
+    connection: VoiceConnection,
+    dispatcher: CoroutineDispatcher
+) : CoroutineScope {
+    override val coroutineContext: CoroutineContext =
+        SupervisorJob() + dispatcher + CoroutineName("Voice Connection Incoming Streams")
+
+    // this will be set before it is used as the key is received before the udp connection is even established
+    internal lateinit var key: ByteArray
```

Comment:
This is not thread safe, there is no happens-before relationship between setting and using `key`.
-> `@Volatile` or `atomic`?